src/components/routers/SubflowRouter.tsx

```diff
@@ -109,24 +108,15 @@ export default class SubflowRouter extends React.PureComponent<SubflowRouterForm
             return this.props.getExitTranslations();
         }
 
-        let flowName: string = '';
-        let flowUUID: string = '';
-
-        if (this.props.action) {
-            if (this.props.action.type === 'start_flow') {
-                ({ flow_name: flowName, flow_uuid: flowUUID } = this.props.action as StartFlow);
-            }
-        }
-
         return (
             <div>
                 <p>Select a flow to run</p>
                 <FlowElement
                     ref={this.props.onBindWidget}
                     name="Flow"
                     endpoint={this.context.endpoints.flows}
-                    flow_name={flowName}
-                    flow_uuid={flowUUID}
+                    flow_name={(this.props.action as StartFlow).flow_name}
+                    flow_uuid={(this.props.action as StartFlow).flow_uuid}
```

Comment:
Having two casts here feels odd to me. Since we know we have a StartFlow action, maybe we make that a first class thing that's typed?

Reply:
Ah right, that's an oversight on my part. Changing the `action` prop in the `SubflowRouterFormProps` interface to `StartFlow` removes the need to cast there.